app-experimental.tex

```diff
@@ -2395,3 +2395,222 @@ \section{Distributed Capabilities For Peripherals And Accelerators} % <<<
 % >>>
 
 % vim: foldmethod=marker:foldmarker=<<<,>>>
+
+% >>>
+% >>>
+\section{Compartment ID Sealing} % <<<
+
+\subsection{Motivation}
+
+Compartment identification is essential. Code and data associated with a compartment must be somehow identified and validated when they are being accessed.
+The most natural way seems to be to give them identification numbers.
+This has traditionally been done in software, e.g., processes are assigned process IDs (PIDs) in operating systems. In some cases, a software ID can correspond to a hardware ID, e.g., PIDs can correspond to address space identifiers (ASIDs).
+However, that is not necessarily the case.
+In this proposal, we suggest an architectural ID for compartments.
+We call these numbers compartment IDs (CIDs). That means that one CID is mapped to one compartment at a time.
+An architecturally defined CID needs to be implemented by hardware.
+
+\subsection{Storage}
+\label{subsec:storage}
+
+The first question to tackle is where to store a CID. We envision multiple options for that.
+First, one can use a dedicated register for compartment identification as done by the Morello architecture.
+Second, one can import CIDs into the capability format, which is what we propose in this document.
+Having this solution leads to an atomic change of CID and code capability -- a property we envision to be useful for secure compartmentalisation.
+
+The CID will substitute the otype bits (18 bits) in the 128-bit capability format.
+The otype bits are currently not well-used. The otype field indicates whether a capability is sealed and if so which “type” it has.
```

Comment:
unicode quotes are evil: ``type''

Reply:
Ah you are right. That comes from copying things out of the google doc...